src/mon/OSDMonitor.cc

```diff
@@ -1364,6 +1365,22 @@ bool OSDMonitor::should_propose(double& delay)
     return true;
   }
 
+  // propose as fast as possible if updating up_thru or pg_temp
+  // want to merge OSDMap changes as much as possible
+  if ((pending_inc.new_primary_temp.size() == 1
+      || pending_inc.new_up_thru.size() == 1)
+      && pending_inc.new_state.size() < 2) {
```

Comment:
@mslovy why we are checking the `new_state.size()` here? just to minimize the impact of flapping osd? and only let the single osd change pass through? am i right?